src/libraries/RegistryHashes.sol

```diff
@@ -3,44 +3,304 @@ pragma solidity ^0.8.20;
 
 import {IDelegateRegistry} from "../IDelegateRegistry.sol";
 
+/**
+ * @title Library for calculating the hashes and storage locations used in the delegate registry
+ *
+ * The encoding for the 5 types of delegate registry hashes should be as follows
+ *
+ * ALL:         keccak256(abi.encode(from, rights, to))
+ * CONTRACT:    keccak256(abi.encode(from, rights, to, contract_))
+ * ERC721:      keccak256(abi.encode(from, rights, to, tokenId, contract_))
+ * ERC20:       keccak256(abi.encode(from, rights, to, contract_))
+ * ERC1155:     keccak256(abi.encode(from, rights, to, tokenId, contract_))
```

Comment:
Why invert the tokenId & contract order instead of having contract consistently in the fourth hash slot across all (non-ALL) types?

Reply:
The non-sequential input types make it slightly harder to put in bad inputs. e.g compiler error or front end abi error